ledger/src/blockstore.rs

```diff
@@ -1419,9 +1409,9 @@ impl Blockstore {
         Ok(newly_completed_data_sets)
     }
 
-    fn should_insert_coding_shred(shred: &Shred, last_root: &RwLock<u64>) -> bool {
+    fn should_insert_coding_shred(shred: &Shred, max_root: Slot) -> bool {
         debug_assert_matches!(shred.sanitize(), Ok(()));
-        shred.is_code() && shred.slot() > *last_root.read().unwrap()
+        shred.is_code() && shred.slot() > max_root
```

Comment:
There is a slight change of behavior here, because previously the code was fetching the most recent `last_root` by passing the `&RwLock<u64>` directly here, but now the `max_root` is loaded earlier in the call stack and can be stalled and not be equal to `self.max_root()` anymore.

This can potentially result in more unnecessary processing of old shreds, but I don't know how much impact it has.

Is there a downside to pass `&AtomicU64` here and do the `load` here as opposed to earlier in the call-tack?

Reply:
> but now the max_root is loaded earlier in the call stack and can be stalled and not be equal to self.max_root() anymore.

> This can potentially result in more unnecessary processing of old shreds, but I don't know how much impact it has.

`Blockstore::check_insert_coding_shred()` is the caller and the new location of `self.max_root()`:
https://github.com/solana-labs/solana/blob/04e4efd8ae88005572c00fa60d329101c5a3e1de/ledger/src/blockstore.rs#L1185

And here is the full implementation of `should_insert_coding_shred()`:
https://github.com/solana-labs/solana/blob/04e4efd8ae88005572c00fa60d329101c5a3e1de/ledger/src/blockstore.rs#L1412-L1415

So, the change is that `self.max_root()` is getting called prior to `shred.is_code() && shred.slot()`:
-`shred.is_code()` is a simple equality check on shred type
- `shred.slot()` just reference the slot from the shred's common header.

So, while it is getting called earlier in the callstack, the amount earlier is these two minimal operations.